gapic/src/main/com/google/gapid/perfetto/views/State.java

```diff
@@ -68,6 +73,9 @@
   private final AtomicInteger lastSelectionUpdateId = new AtomicInteger(0);
   private HashMultimap<Long, Long> selectedThreads;     // upid -> utids
   private TimeSpan highlight = TimeSpan.ZERO;
+  private TreeMap<Long, Boolean> flags;
+  private boolean flagHovered = false;
+  private double flagHoverXpos;
```

Comment:
Good idea. Matter of fact, I think the entire flag related stuff (including the TreeMap) can be moved to the RootPanel. I put the TreeMap here in State for the first version of the flag feature, since it involved using the TimelinePanel's onMouseMove() and RootPanel's rendering. But now, given that the entire flag mechanism happens within the RootPanel, I think its better to move everything there.